cli_tools/gce_windows_upgrade/upgrader/validators.go

```diff
@@ -200,21 +210,23 @@ func validateAndDeriveOSDisk(osDisk *compute.AttachedDisk, derivedVars *derivedV
 	return nil
 }
 
-func validateLicense(osDisk *compute.AttachedDisk, sourceOS string) error {
+func validateLicense(osDisk *compute.AttachedDisk, sourceOS, targetOS string) error {
 	matchSourceOSVersion := false
 	upgraded := false
 	for _, lic := range osDisk.Licenses {
-		if strings.HasSuffix(lic, expectedCurrentLicense[sourceOS]) {
-			matchSourceOSVersion = true
-		} else if strings.HasSuffix(lic, licenseToAdd[sourceOS]) {
-			upgraded = true
+		for _, expectedLic := range upgradePaths[sourceOS][targetOS].expectedCurrentLicense {
+			if strings.HasSuffix(lic, expectedLic) {
+				matchSourceOSVersion = true
+			} else if strings.HasSuffix(lic, upgradePaths[sourceOS][targetOS].licenseToAdd) {
+				upgraded = true
+			}
 		}
 	}
 	if !matchSourceOSVersion {
-		return daisy.Errf(fmt.Sprintf("Can only upgrade GCE instance with %v license attached", expectedCurrentLicense[sourceOS]))
+		return daisy.Errf(fmt.Sprintf("Can only upgrade GCE instance with any of %v license attached", upgradePaths[sourceOS][targetOS].expectedCurrentLicense))
```

Comment:
I'd suggest a clearer error message.

No valid Windows Server PayG license can be found. Any of the following licenses are required %v"

Reply:
done